llvm/lib/CodeGen/AtomicExpandPass.cpp

```diff
@@ -1770,7 +1800,7 @@ bool AtomicExpand::expandAtomicOpToLibcall(
   auto PtrTypeAS = PointerOperand->getType()->getPointerAddressSpace();
   Value *PtrVal = Builder.CreateBitCast(PointerOperand,
                                         Type::getInt8PtrTy(Ctx, PtrTypeAS));
-  unsigned PtrValAS = DL.getAllocaAddrSpace();
+  auto PtrValAS = PointerOperandIsCap ? PtrTypeAS : DL.getAllocaAddrSpace();
```

Comment:
When would this ever not be just PtrTypeAS?

Reply:
I think this could happen for other architectures, where you can only pass one type of pointer to libfuncs. We can pass both AS0 and AS200, but that might not always be possible.